### DISCUSSION TITLE
[Curso] Fundamentos do Aspnet 6 [Aula] Iniciando Cadastro de Usuário

### DISCUSSION THREAD

**Discussion by paulamora-is** [balta-io/forum#365] [Tira Dúvidas] [answered]
Olá, Balta,tudo bem? Eu tenho duas dúvidas com o tópico "Autorização e Autenticação", na aula Iniciando Cadastro de Usuário

**A primeira dúvida:**

Quando mudei as informações do mapeamento de User, no momento que rodei a migration , recebi esse warning. O que seria?

![image](https://github.com/balta-io/forum/assets/55562741/4ccf236d-5804-4db7-b325-bda363843bfe)

E minha migration gerou assim: 

```
public partial class NewUserMap : Migration
    {
        /// <inheritdoc />
        protected override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.AlterColumn<string>(
                name: "PasswordHash",
                table: "Users",
                type: "nvarchar(255)",
                maxLength: 255,
                nullable: false,
                defaultValue: "",
                oldClrType: typeof(string),
                oldType: "nvarchar(max)",
                oldNullable: true);

            migrationBuilder.AlterColumn<string>(
                name: "Email",
                table: "Users",
                type: "VARCHAR(160)",
                maxLength: 160,
                nullable: false,
                defaultValue: "",
                oldClrType: typeof(string),
                oldType: "nvarchar(max)",
                oldNullable: true);
        }

        /// <inheritdoc />
        protected override void Down(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.AlterColumn<string>(
                name: "PasswordHash",
                table: "Users",
                type: "nvarchar(max)",
                nullable: true,
                oldClrType: typeof(string),
                oldType: "nvarchar(255)",
                oldMaxLength: 255);

            migrationBuilder.AlterColumn<string>(
                name: "Email",
                table: "Users",
                type: "nvarchar(max)",
                nullable: true,
                oldClrType: typeof(string),
                oldType: "VARCHAR(160)",
                oldMaxLength: 160);
        }
    }

```

No entanto, no momento de dar Update-Database recebi  a exceção: **Microsoft.Data.SqlClient.SqlException (0x80131904): There is already an object named 'Categories' in the database.**

Para atualizar apenas um mapeamento e rodar uma nova migration, eu preciso comentar os outros mapeamentos para não rodar tudo? Como evito esse problema?

**A segunda dúvida**
Quando crio a rota v1/account, no método Register, na condição que verifica se a ModelState é válida, eu recebo o seguinte erro:

![image](https://github.com/balta-io/forum/assets/55562741/09254f80-e192-4a89-a553-86ea051f3bdd)

Pode me ajudar com essas questões? Obrigada.

**Comment by andrebaltieri** [comment, marked as answer]
@paulamora-is bom dia!

Sobre o warning, sempre que rodamos uma operação scaffold (Que gera código) pode ocorrer dele sobrescrever algum arquivo, então realmente temos que ter cuidado.

Sobre seu erro no migration, ele ocorre pois você já tem uma tabela com este nome (Categories) no banco. Recomendo que faça todos os cursos da carreira, incluindo o de SQL, vale muito a pena e ajuda muito nestes detalhes.

Sobre os updates, eles são cumulativos, como explico no curso de EF. Ele usa a tabela Migration History para saber quais foram aplicados e não aplicar novamente.

Sobre o último erro, preciso ver o código do seu RegisterViewModel.

Uma última recomendação... faça todos os cursos da carreira, ou se estiver seguindo as novas carreiras, faça-as na ordem que aparecem no site... vai te dar uma ótima bagagem e quando chegar no ASP.NET já vai estar bem treinada em EF hahahah!!

**Comment by malighieri** [comment]
Complementando a explicaçào do Warning, acredito que o principal foco dele
é te alertar que sua alteração no modelo pode ocasionar perda de dados. O
que explica isso é o fato de você ter alterado as duas colunas da tabela
Users, PasswordHash e Email, passando o tamanho de *nvarchar(Max)* para
*nvarchar(255)* e *nvarchar(160)* respectivamente. Caso a tabela já tenha
dados, como o *nvarchar(Max)* suporta mais caracteres, poderia ocorrer
perda ao mudar o tamanho para *nvarchar(255)* ou *nvarchar(160)*. Por isso
o warning pede pra você analisar mais de perto a migration gerada para
evitar isso.

Miron Alighieri dos Santos


Em seg., 17 de jul. de 2023 às 09:39, Andre Baltieri <
***@***.***> escreveu:
